{"tags":["c#","исключения"],"owner":{"user_type":"does_not_exist","display_name":"user358283"},"is_answered":true,"view_count":51,"accepted_answer_id":1046572,"answer_count":1,"score":0,"last_activity_date":1573825632,"creation_date":1573823860,"last_edit_date":1573825632,"question_id":1046558,"link":"https://ru.stackoverflow.com/questions/1046558/%d0%9f%d0%b5%d1%80%d0%b5%d1%85%d0%b2%d0%b0%d1%82-%d0%b8%d1%81%d0%ba%d0%bb%d1%8e%d1%87%d0%b5%d0%bd%d0%b8%d1%8f-%d1%81%d1%82%d0%be%d0%b8%d1%82-%d0%bb%d0%b8-c","title":"Перехват исключения (стоит ли) C#","body":"<p>Мне кажется код немного кривоват, но не могу понять как лучше его переделать.\nКод:</p>\n\n<pre><code>void Button1Click(object sender, EventArgs e)\n{\n    if (openFileDialog1.ShowDialog() == DialogResult.OK) \n    {\n        if (!System.IO.File.Exists(openFileDialog1.FileName))\n        {\n            throw new PathException(\"Некорректный путь к файлу\", openFileDialog1.FileName);\n        }\n        try\n        {\n            textBox7.Text = openFileDialog1.FileName;\n        }\n        catch (PathException ex)\n        {\n            MessageBox.Show(String.Format(\"{0}: {1}\", ex.Message, ex.PathValue), \"Ошибка.\", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);\n        }\n    }       \n}\n</code></pre>\n\n<p>Класс исключения:</p>\n\n<pre><code>public class PathException : ArgumentException\n{   \n    public string PathValue {get; set;}\n    public PathException(string message, string pathValue)\n        :base(message)\n    {\n        PathValue = pathValue;\n    }\n}\n</code></pre>\n\n<p>Что можно исправить? </p>\n"}